WcaOnRails/app/javascript/edit-schedule/EditSchedule.jsx

```diff
@@ -0,0 +1,218 @@
+      id: competitionInfo.id,
+      schedule: competitionInfo.scheduleWcif,
+    };
+
+    this.setState({ saving: true });
+    promiseSaveWcif(wcif).then(response => {
+      return Promise.all([response, response.json()]);
+    }).then(([response, json]) => {
+      if(!response.ok) {
+        throw new Error(`${response.status}: ${response.statusText}\n${json["error"]}`);
+      }
+      this.setState({ savedScheduleWcif: _.cloneDeep(competitionInfo.scheduleWcif), saving: false });
+    }).catch(e => {
+      this.setState({ saving: false });
+      alert(`Something went wrong while saving.\n${e.message}`);
+    });
+  }
+
+
+  unsavedChanges() {
+    return !_.isEqual(this.state.savedScheduleWcif, this.props.competitionInfo.scheduleWcif);
+  }
+
+  onUnload = e => {
+    // Prompt the user before letting them navigate away from this page with unsaved changes.
+    if(this.unsavedChanges()) {
+      let confirmationMessage = "You have unsaved changes, are you sure you want to leave?";
+      e.returnValue = confirmationMessage;
+      return confirmationMessage;
+    }
+  }
+
+  componentDidMount() {
+    wca.datetimepicker();
+    window.addEventListener("beforeunload", this.onUnload);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("beforeunload", this.onUnload);
+  }
+
+  render() {
+    let { competitionInfo, locale } = this.props;
+    let scheduleWcif = competitionInfo.scheduleWcif;
+
+    let actionsHandlers = {
+      addVenue: e => {
+        e.preventDefault();
+        addVenueToSchedule(competitionInfo);
+        rootRender();
+      },
+      removeVenue: (e, index) => {
+        e.preventDefault();
+        if (!confirm(`Are you sure you want to remove the venue "${scheduleWcif.venues[index].name}" and all the associated rooms and schedules?`)) {
+          return;
+        }
+        scheduleWcif.venues.splice(index, 1);
+        rootRender();
+      },
+    };
+
+    let isThereAnyRoom = scheduleWcif.venues.some(function (element) {
+      return element.rooms.length > 0;
+    });
+
+    let defaultActivePanel = isThereAnyRoom ? "2" : "1";
+
+    let unsavedChanges = null;
+    if (this.unsavedChanges()) {
+      unsavedChanges = <UnsavedChangesAlert
+                           unsavedChanges={this.unsavedChanges()}
+                           actionHandler={this.save}
+                           saving={this.state.saving}
+                       />
+    }
+
+    return (
+      <div>
+        {unsavedChanges}
+        <Row>
+          <IntroductionMessage />
+          <Col xs={12}>
+            <PanelGroup accordion id="accordion-schedule" defaultActiveKey={defaultActivePanel}>
+              <Panel id="venues-edit-panel" bsStyle="info" eventKey="1">
+                <div id="accordion-schedule-heading-1" className="panel-heading heading-as-link" aria-controls="accordion-schedule-body-1" role="button" data-toggle="collapse" data-target="#accordion-schedule-body-1" data-parent="#accordion-schedule">
```

Comment:
I'm surprised to see those `data` attributes adding behaviour instead of some React way. Looking at the docs [here](https://react-bootstrap.github.io/components/panel/#panels-accordion) I can see `toggle` attribute on `Panel.Title` that seems to do the job without any additional `div` =)